ocs/cluster.py

```diff
@@ -0,0 +1,248 @@
+"""
+A module for all rook functionalities and abstractions.
+
+This module has rook related classes, support for functionalities to work with
+rook cluster. This works with assumptions that an OCP cluster is already
+functional and proper configurations are made for interaction.
+"""
+
+import logging
+import os
+import base64
+
+import oc.openshift_ops as ac
+import resources.pod as pod
+from resources import ocs
+import ocs.defaults as default
+from utility.templating import generate_yaml_from_jinja2_template_with_data
+from ocsci.config import ENV_DATA
+from ocs import ocp
+from ocs.exceptions import *
+
+POD = ocp.OCP(kind='Pod', namespace=ENV_DATA['cluster_namespace'])
+CEPHCLUSTER = ocp.OCP(
+    kind='CephCluster', namespace=ENV_DATA['cluster_namespace']
+)
+
+logger = logging.getLogger(__name__)
+
+
+class CephCluster(object):
+    """
+    Handles all cluster related operations from ceph perspective
+
+    This class has depiction of ceph cluster. Contains references to
+    pod objects which represents ceph cluster entities.
+
+    Attributes:
+        _ocs_pods (list) : A list of  ceph cluster related pods
+        _cluster_name (str): Name of ceph cluster
+        _namespace (str): openshift Namespace where this cluster lives
+        _api_client (APIClient): api-client used for interacting with openshift
+        _url_prefix (str): url prefix for REST api interaction
+    """
+
+    def __init__(self, **config):
+        # cluster_name is name of cluster in rook of type CephCluster
+        self._cluster_name = config.get('cluster_name', 'rook-ceph')
+        """
+        TODO: Instantiate CephCluster object somewhere higher after deployment
+        Read namespace and cluster name from config.
+        """
+        self._namespace = config.get(
+            'namespace', ENV_DATA['cluster_namespace']
+        )
+        self._ocs_pods = pod.get_all_pods(self._namespace)
+
+        #self.cluster_resource = os.path.join(
+        #    default.TEMPLATE_DIR, "cluster.yaml")
+        # Once we figure out where to initialize this CephCluster object we
+        # will pass **config
+        #self.cluster_config = generate_yaml_from_jinja2_template_with_data(
+        #    self.cluster_resource, **config
+        #)
+        self.cluster_config = CEPHCLUSTER.get()['items'][0]
```

Comment:
```python
= CEPHCLUSTER.get().get('items')[0]
```

Reply:
self.cluster_yaml_config to differentiate from ceph_conf config?

Reply:
Done